File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/service/ScheduleDomainService.java
package com.example.chat.domain.service;

import java.time.Instant;
import java.time.temporal.ChronoUnit;

import com.example.chat.domain.message.Message;
import com.example.chat.domain.schedule.CronExpression;
import com.example.chat.domain.schedule.ScheduleRule;
import com.example.chat.domain.schedule.ScheduleType;

/**
 * 테스트에서 사용되는 ScheduleDomainService의 최소 구현
 */
public class ScheduleDomainService {

    public ScheduleRule createOneTimeSchedule(Message message, Instant scheduledAt) {
        if (scheduledAt == null) {
            throw new IllegalArgumentException("Scheduled time cannot be null");
        }
        Instant now = Instant.now();
        if (!scheduledAt.isAfter(now)) {
            throw new IllegalArgumentException("Scheduled time must be in the future");
        }
        if (scheduledAt.isAfter(now.plus(365, ChronoUnit.DAYS))) {
            throw new IllegalArgumentException("Scheduled time cannot be more than 1 year in the future");
        }

        return new ScheduleRule(ScheduleType.ONE_TIME, message, scheduledAt, null);
    }

    public ScheduleRule createRecurringSchedule(Message message, CronExpression cronExpression) {
        if (cronExpression == null) {
            throw new IllegalArgumentException("Cron expression cannot be null");
        }
        return new ScheduleRule(ScheduleType.RECURRING, message, null, cronExpression);
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/service/ScheduleDomainService.kt
package com.example.chat.domain.service

import com.example.chat.domain.message.Message
import com.example.chat.domain.schedule.CronExpression
import com.example.chat.domain.schedule.ScheduleRule
import com.example.chat.domain.schedule.ScheduleType
import java.time.Instant
import java.time.temporal.ChronoUnit

/**
 * 테스트에서 사용되는 ScheduleDomainService의 최소 구현
 */
class ScheduleDomainService {

    fun createOneTimeSchedule(message: Message, scheduledAt: Instant?): ScheduleRule {
        requireNotNull(scheduledAt) { "Scheduled time cannot be null" }
        val now = Instant.now()
        require(scheduledAt.isAfter(now)) { "Scheduled time must be in the future" }
        require(!scheduledAt.isAfter(now.plus(365, ChronoUnit.DAYS))) { "Scheduled time cannot be more than 1 year in the future" }

        return ScheduleRule(ScheduleType.ONE_TIME, message, scheduledAt, null)
    }

    fun createRecurringSchedule(message: Message, cronExpression: CronExpression?): ScheduleRule {
        requireNotNull(cronExpression) { "Cron expression cannot be null" }
        return ScheduleRule(ScheduleType.RECURRING, message, null, cronExpression)
    }
}
